### DISCUSSION TITLE
Why don't output descriptors live in `rust-bitcoin`

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#2027] [Q&A] [answered]
Why are output descriptors defined in `rust-miniscript` instead of in `rust-bitcoin`? They seem like a general wallet layer concept?

**Comment by apoelstra** [comment, marked as answer]
Because descriptors and miniscript are two parts of the same project, which is much larger than anything else in rust-bitcoin.

**Reply by tcharding** [reply]
For anyone else reading this thread the following old blog post helps explain the comment above.

https://medium.com/blockstream/miniscript-bitcoin-scripting-3aeff3853620

**Reply by stevenroose** [reply]
Yeah, I would mark this as answered. I'll do that, change it back if I'm mistaken.

**Reply by tcharding** [reply]
Cheers!

**Comment by tcharding** [comment]
What do you mean by two parts of the same project? Greppnig for `desciptor::` it seems only `policy` and `psbt` depend on descriptors, isn't an implementation of output descriptors a stand alone thing?

**Reply by tcharding** [reply]
Is it because of lifting that they are intertwined?

**Reply by tcharding** [reply]
Oh don't mind me, they all contain a `Miniscript` struct.

**Reply by apoelstra** [reply]
There is conceptually no difference between "descriptors" and "miniscript". Miniscript is the part of descriptors that contains the actual script, but users don't need to know or care about the distinction.

I'm not sure what you mean by "only `policy` and `psbt` depend on descriptors". Descriptors are the main thing that rust-miniscript implements.

**Comment by tcharding** [comment]
Issue with discussion on this topic: https://github.com/rust-bitcoin/rust-bitcoin/issues/3695

**Comment by apoelstra** [comment]
I think this one can be closed -- it was a straightforward question about moving descriptor types into rust-bitcoin (which is a no-go because of how big the descriptors project is), not a more nuanced one about stuff we're actually going to do (pulling `DescriptorPublicKey` into a crate that `bitcoin`/`addresses` uses, attempting to invert the bitcoin/miniscript dep, etc).

**Reply by apoelstra** [reply]
See https://github.com/rust-bitcoin/rust-bitcoin/discussions/4862 -- apparently we don't want to close resolved Q&A; just leave them "answered".
